pennylane/beta/tapes/tape.py

```diff
@@ -833,3 +835,105 @@ def data(self):
     @data.setter
     def data(self, params):
         self.set_parameters(params, trainable_only=False)
+
+    # ========================================================
+    # execution methods
+    # ========================================================
+
+    def execute(self, device, params=None):
+        """Execute the tape on a quantum device.
+
+        Args:
+            device (~.Device): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): The quantum tape operation parameters. If not provided,
+                the current tape parameters are used (via :meth:`~.get_parameters`).
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.beta.tapes import QuantumTape
+            from pennylane.beta.queuing import expval, var, sample, probs
+
+            with QuantumTape() as tape:
+                qml.RX(0.432, wires=0)
+                qml.RY(0.543, wires=0)
+                qml.CNOT(wires=[0, 'a'])
+                qml.RX(0.133, wires='a')
+                probs(wires=[0, 'a'])
+
+        If parameters are not provided, the existing tape parameters are used:
+
+        >>> dev = qml.device("default.qubit", wires=[0, 'a'])
+        >>> tape.execute(dev)
+        array([[8.84828969e-01, 3.92449987e-03, 4.91235209e-04, 1.10755296e-01]])
+
+        Parameters can be optionally passed during execution:
+
+        >>> tape.execute(dev, params=[1.0, 0.0, 1.0])
```

Comment:
I think it's obvious, but just wanted to ask the question: is it clear that the parameters of a tape don't include the choice of wires?

Reply:
I thought it was, but maybe it's not? tape parameters == tape gate parameters. They can be marked trainable or non-trainable.

Reply:
Yeah I think it's ok, I just noticed it's different from writing a quantum function, whose signature can include the choice of wires as "parameters" :thinking: